### DISCUSSION TITLE
Help me expected primary-expression before ‘==’ token

### DISCUSSION THREAD

**Discussion by 032321811za** [arendst/Tasmota#17730] [Q&A]
expected primary-expression before ‘==’ token
code
#include <ESP8266WiFi.h>
#include <TridentTD_LineNotify.h>

const int SensorPin = D6;
int SensorPin1 = D7;
int SensorState = 0;

#define SSID “OnePlus 7T” //Use Wifi
#define PASSWORD “032321811” //Pass Wifi
#define LINE_TOKEN “jEC2AdLjUJ4n2ftxKpJTNUdkeYCRyaUsDmZrjr03uU6” //Token Line

void setup() {
Serial.begin(115200); Serial.println();
pinMode(SensorPin, INPUT);
pinMode(SensorPin1, INPUT);
Serial.println(LINE.getVersion());

WiFi.begin(SSID, PASSWORD);
Serial.printf(“WiFi connecting to %s\n”, SSID);
while (WiFi.status() != WL_CONNECTED) {
Serial.print(“.”);
delay(400);
}
Serial.printf("\nWiFi connected\nIP : ");
Serial.println(WiFi.localIP());

// กำหนด Line Token
LINE.setToken(LINE_TOKEN);

// ตัวอย่างส่งข้อความ
LINE.notify(“เชื่อมต่อ สำเร็จ”);
}

void loop() {

digitalWrite(SensorPin); {
if (SensorPin == HIGH)
//while (digitalRead(SensorPin); == HIGH) delay(10);
Serial.println(“Enter !”);
LINE.notify(“ไฟไหม้บ้านแล้ว”);
}
delay(10);
}

void loop() {

Text

digitalWrite(SensorPin1); {
if (SensorPin1 == HIGH)
//while (digitalRead(SensorPin1); == HIGH) delay(10);
Serial.println(“Enter !”);
LINE.notify(“ควันเต็มบ้านแล้ว”);
}
delay(10);
}
![image](https://user-images.githubusercontent.com/122876323/212878603-508a57b2-c998-4d42-95a9-d3d968a0206c.png)
![image](https://user-images.githubusercontent.com/122876323/212878683-a2691973-076e-48d6-b1c3-d48479fe4bc6.png)

**Comment by barbudor** [comment]
Have you read the documentation?
https://tasmota.github.io/docs/Compile-your-build/

You are not using a supported tool

**Reply by sfromis** [reply]
Does not at all look Tasmota-related, just some random beginner-trouble with coding in Arduino IDE. The screenshot posted indicates that you need two arguments for digitalWrite, both pin and value. And better find a place where you are not badly off topic.
